src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Filters/ResourceUrlFilter.cs

```diff
@@ -35,10 +35,12 @@ public ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArguments argu
                 resourcePath = _httpContextAccessor.HttpContext.Request.PathBase.Add(resourcePath.Substring(1)).Value;
             }
 
-            // Don't prefix cdn if the path is absolute, or is in debug mode.
+            // Don't prefix cdn if the path includes a protocol, i.e. is an external url, or is in debug mode.
             if (!_options.DebugMode
                 && !String.IsNullOrEmpty(_options.CdnBaseUrl)
-                && !Uri.TryCreate(resourcePath, UriKind.Absolute, out var uri))
+                && !resourcePath.StartsWith("https://", StringComparison.OrdinalIgnoreCase)
+                && !resourcePath.StartsWith("http://", StringComparison.OrdinalIgnoreCase)
+                && !resourcePath.StartsWith("//", StringComparison.OrdinalIgnoreCase))
```

Comment:
@deanmarcussen 

Just tried, it seems that `System.Uri.TryCreate()` does the same checkings

It returns true for `System.Uri.TryCreate("https://toto.com")`

And it also returns true for `System.Uri.TryCreate("//toto")`

Reply:
@jtkech but it also returns true for `/foo.js` 

That was the problem :(

Unless there is a better way? I tried with `IsWellFormed` as well, but wasn't correct.

Reply:
@deanmarcussen No, just tried, return false in that case ;)

Reply:
https://dotnetfiddle.net/t6gb6x

Which `UriKind` @jtkech ?

returns true with UriKind.Absolute for me

Reply:
if (System.Uri.TryCreate("/foo.js", System.UriKind.Absolute, out var uri))
            {
                ; // <= Doesn't go there
            }

Reply:
Yes, under fiddle it creates `file:///foo.js` ???

Reply:
Hmm. Interesting. 

`file:///foo.js`
 
That's what I get on the mac as well. 

Did we find a bug?

Reply:
Yes maybe ;)

And your PR would be a good workaround

I'll take a look at the source code

Reply:
@deanmarcussen 

Okay it's because, e.g. some grabbed comments

> A relative Uri wins over implicit Unix path unless uriKind == Absolute

> In .NET new Uri ("/foo", UriKind.RelativeOrAbsolute) returns a relative Uri whereas Mono assumes it is an absolute file path.

> Mono diverges from .NET because UNIX file paths starting with ‘/’ are absolute file paths.

So your fix is right

Reply:
ok, interesting, and of course that makes sense.

There's tests now on these, so it should fail on the CI if we tried to change it.